//
// Definitions for the struct map testcases.
//
// Copyright 2009 Simon Kallweit
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

package test;

message StructTest {

  enum TestEnum {
    VALUE1 = 1;
    VALUE2 = 2;
    VALUE3 = 3;
  }

  required int32 field_int32 = 1;
  required int64 field_int64 = 2;
  required bool field_bool = 3;
  required TestEnum field_enum = 4;
  required string field_string = 5;
  required bytes field_bytes = 6;
  
  message Nested1 {
    required int32 field_int32 = 1;
    required int64 field_int64 = 2;
  }
  
  required Nested1 nested1 = 7;
  
  message Nested2 {
    required string field_string = 1;
  }
  
  repeated Nested2 nested2 = 8;
}
